src/rgw/rgw_common.h

```diff
@@ -1917,6 +1934,7 @@ struct rgw_obj {
   rgw_obj_key key;
 
   bool in_extra_data{false}; /* in-memory only member, does not serialize */
+  bool head_obj{true}; /* in-memory only member, does not serialize */
```

Comment:
@mattbenjamin @mikulely 

please see the update codes, I had changed the schem according @yehudasa 's comments.